uast/transformer/positioner/positions_test.go

```diff
@@ -180,3 +194,55 @@ a3`
 		})
 	}
 }
+
+func TestPosIndexUTF16(t *testing.T) {
+	// Verify that a UTF-16 code point offset -> byte offset conversion works.
+	const source = `line1
```

Comment:
Please mention also in the comment that this case includes surrogate pairs, which are an important corner case for UTF-16.